### PR TITLE
perf(ui): Fix nested Proxies and recursive proxy calls

### PR DESCRIPTION
**Pull request by billyvg**
Optimized usage of `MetaProxy` in group details.

Fixes ISSUE-205

### CODE REVIEW

File: tests/js/spec/components/events/interfaces/utils.spec.jsx [hunk 1]
```diff
@@ -76,6 +77,40 @@ describe('components/interfaces/utils', function() {
...
+        url: 'https://www.sentry.io',
+        query: '',
+        data: null,
+        method: 'GET',
+      };
+      const eventWithProxy = withMeta(data);
+      getCurlCommand(eventWithProxy);
+
+      // This may need to change, but we should aim to keep this low
+      expect(spy).toHaveBeenCalledTimes(172);
```
Comment (markstory) [inline tests/js/spec/components/events/interfaces/utils.spec.jsx]: Should this scale with the number of properties in the proxied object graph?

File: src/sentry/static/sentry/app/components/events/meta/metaProxy.jsx [hunk 2]
```diff
@@ -29,34 +30,50 @@ export class MetaProxy {
...
 
 export function withMeta(event) {
-  if (!event) return null;
+  if (!event) {
+    return null;
+  }
 
   // Return unproxied `event` if browser does not support `Proxy`
-  if (typeof window.Proxy === 'undefined') return event;
+  if (typeof window.Proxy === 'undefined' || typeof window.Reflect === 'undefined') {
```
Comment (jan-auer) [inline src/sentry/static/sentry/app/components/events/meta/metaProxy.jsx]: Are we going to polyfill this somehow? We might make changes soon that might require rendering meta data in certain places to show errors to the user.

**Review by jan-auer — APPROVED** [review]
Thanks, @billyvg !

File: src/sentry/static/sentry/app/components/events/meta/metaProxy.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/events/meta/metaProxy.jsx]: Let's polyfill when we're ready to make those changes, we'll have to refactor Reflect out, but should be simple.

File: tests/js/spec/components/events/interfaces/utils.spec.jsx (hunk 1 above)
Comment (billyvg) [reply tests/js/spec/components/events/interfaces/utils.spec.jsx]: It does, but not linearly - I just wanted to capture a baseline for the number of times this is called (before optimization it was about 6-10x more)

**Pull request merged**